lib/beaker-puppet/install_utils/foss_utils.rb

```diff
@@ -963,9 +963,27 @@ def install_repo_configs(host, buildserver_url, package_name, build_version, cop
           repo_config_folder_url = "%s/%s/%s/repo_configs/%s/" %
             [ buildserver_url, package_name, build_version, host.repo_type ]
 
-          repo = fetch_http_file( repo_config_folder_url,
-                                  repo_filename,
-                                  copy_dir )
+          repo_config_url = "#{ repo_config_folder_url }/#{ repo_filename }"
+          install_repo_configs_from_url( host, repo_config_url, copy_dir )
+        end
+
+        # Installs the repo configs on a given host
+        #
+        # @param [Beaker::Host] host Host to install configs on
+        # @param [String] repo_config_url URL to the repo configs
+        # @param [String] copy_dir Local directory to fetch files into & SCP out of
+        #
+        # @return nil
+        def install_repo_configs_from_url(host, repo_config_url, copy_dir = nil)
+          copy_dir ||= Dir.mktmpdir
+          repoconfig_filename = File.basename(  repo_config_url )
+          repoconfig_folder   = File.dirname(   repo_config_url )
+
+          repo = fetch_http_file(
```

Comment:
@melissa I'm wary of degrading to a different install if the repo_configs aren't available. That's something that we've done historically, and we've had two problems with that approach: transparency & debugging/maintenance.

One is that magic changes underneath a user (like an auto install-method switch) is much harder to debug & understand. The only way to be transparent about degrading in this fashion that we've seen is to scream into the console log, but then we face the problem of screaming wolf too many times & nobody pays attention to the log colors/warnings/errors anymore.

Another is that the more avenues we use to install, the harder it is to maintain that code. If the issue is potentially in the packaging system such that the symptom is that repo_config files aren't being made available, then finding alternate ways to install creates more distance between the cause of errors & the way that we find them. It seems that failing at build-time would be even better, but failing as early as possible during the install step is the best that beaker can do to help a user find a bug like that. 

If we're thinking of transient issues, like network blips over getting the appropriate file rather than the file being in the right place, then there are some things we can definitely do to help with that. Our usual suggestion at this point would be to back off & retry, hoping that those network issues resolve themselves over a small break. Unfortunately, this can hide issues & delay job completion, but it can help overcome things like this. I would say at this point that there's no reason to add that complexity if we don't need it though.